test/api_getProduct_test.dart

```diff
@@ -763,5 +772,86 @@ void main() {
+              PreferenceImportance.ID_NOT_IMPORTANT,
+          () => refreshCounter++,
+        ),
+      );
+      const String languageCode = 'en';
+      const int httpOk = 200;
+      final String importanceUrl =
+          AvailablePreferenceImportances.getUrl(languageCode);
+      final String attributeGroupUrl =
+          AvailableAttributeGroups.getUrl(languageCode);
+      http.Response response;
+      response = await http.get(Uri.parse(importanceUrl));
+      expect(response.statusCode, httpOk);
+      final String preferenceImportancesString = response.body;
+      response = await http.get(Uri.parse(attributeGroupUrl));
+      expect(response.statusCode, httpOk);
+      final String attributeGroupsString = response.body;
+      manager.availableProductPreferences =
+          AvailableProductPreferences.loadFromJSONStrings(
+        preferenceImportancesString,
+        attributeGroupsString,
+      );
+      expect(refreshCounter, 0);
+
+      const String barcode = '0028400047685';
+      final ProductQueryConfiguration configurations =
+          ProductQueryConfiguration(
+        barcode,
+        lc: languageCode,
+        fields: [ProductField.NAME, ProductField.ATTRIBUTE_GROUPS],
+      );
+      final ProductResult result = await OpenFoodAPIClient.getProduct(
+          configurations,
+          user: TestConstants.TEST_USER,
+          queryType: QueryType.TEST);
+      expect(result.status, 1);
+      expect(result.barcode, barcode);
+
+      final String attributeId1 = Attribute.ATTRIBUTE_NUTRISCORE;
+      final String attributeId2 = Attribute.ATTRIBUTE_FOREST_FOOTPRINT;
+      final String importanceId1 = PreferenceImportance.ID_MANDATORY;
+      final String importanceId2 = PreferenceImportance.ID_IMPORTANT;
+      await manager.setImportance(attributeId1, importanceId1);
+      expect(
+          manager.getImportanceIdForAttributeId(attributeId1), importanceId1);
+      expect(refreshCounter, 1);
+      await manager.setImportance(attributeId2, importanceId2);
+      expect(
+          manager.getImportanceIdForAttributeId(attributeId2), importanceId2);
+      expect(refreshCounter, 2);
+      MatchedProduct matchedProduct;
+
+      matchedProduct = MatchedProduct(result.product!, manager);
+      expect(matchedProduct.score, 150);
+      expect(matchedProduct.status, true);
+
+      await manager.setImportance(attributeId1, importanceId2);
+      expect(
+          manager.getImportanceIdForAttributeId(attributeId1), importanceId2);
+      expect(refreshCounter, 3);
+      await manager.setImportance(attributeId2, importanceId1);
+      expect(
+          manager.getImportanceIdForAttributeId(attributeId2), importanceId1);
+      expect(refreshCounter, 4);
+
+      matchedProduct = MatchedProduct(result.product!, manager);
+      expect(matchedProduct.score, 37.5);
+      expect(matchedProduct.status,
+          false); // because FOREST is mandatory, but unknown
```

Comment:
That makes me realize that status has only 2 values. We need 3 to indicate whether products match preferences : "yes", "no", "unknown". There are some products that we know match all preferences, some that do not match at least one required preference, and others for which we don't have enough data.

On the web site we display the 3 categories differently:

![image](https://user-images.githubusercontent.com/8158668/112853149-dce98c00-90ac-11eb-91e8-5ed31a48a880.png)

I'll change the MatchedProduct code

Reply:
Corresponding PR: https://github.com/openfoodfacts/openfoodfacts-dart/pull/142

Reply:
@stephanegigandet Oh, I though `null` was the 3rd choice = "unknown". Isn't it?

Reply:
null indicates that the product has not been matched yet. It's different than unknown.